### PR TITLE
TST: Add tests for sparse quantile/where

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This is the part of https://github.com/pandas-dev/pandas/pull/17386.

### CODE REVIEW

File: pandas/tests/sparse/test_frame.py [hunk 1]
```diff
@@ -1385,3 +1385,391 @@ def test_numpy_func_call(self):
...
+        sparse_expected = SparseDataFrame(dense_expected,
+                                          default_fill_value=other)
+
+        tm.assert_frame_equal(result, dense_expected)
+        tm.assert_sp_frame_equal(result, sparse_expected)
+
+    @pytest.mark.xfail(reason='Wrong SparseBlock initialization '
+                              '(GH 17386)')
+    def test_where_with_int_data_and_bool_other(self):
+        # GH 17386
```
Comment (jreback) [inline pandas/tests/sparse/test_frame.py]: pls pls pls use parametrize. you are repeating way too many things

**Comment by Licht-T** [comment]
@jreback Thanks for your review. Parametrized `other`.

File: pandas/tests/sparse/test_frame.py [hunk 2]
```diff
@@ -1385,3 +1385,209 @@ def test_numpy_func_call(self):
                  'std', 'min', 'max']
         for func in funcs:
             getattr(np, func)(self.frame)
+
+    @pytest.mark.xfail(reason='Wrong SparseBlock initialization '
+                              '(GH 17386)')
```
Comment (jreback) [inline pandas/tests/sparse/test_frame.py]: you need to parameterize on the data itself as well

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by Licht-T** [comment]
@jreback Parametrized `data`.

**Comment by jreback** [comment]
can you rebase

**Comment by Licht-T** [comment]
@jreback Rebased.

**Comment by jreback** [comment]
these tests are failing the CI, not really sure why

**Comment by Licht-T** [comment]
@jreback Okay. I'll check.

**Comment by Licht-T** [comment]
@jreback It seems that segfault happened in `test_where_with_bool_data_and_other`.
Marked `skip`.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Licht-T 

love to have the fixes for these!

**Pull request merged**